code/game/objects/structures/crates_lockers/closets/coffin.dm

```diff
@@ -51,4 +84,69 @@
 	var/atom/A = pick(/obj/landmark/corpse/chef, /obj/landmark/corpse/doctor, /obj/landmark/corpse/engineer, /obj/landmark/corpse/engineer/rig, /obj/landmark/corpse/clown, \
 	/obj/landmark/corpse/scientist, /obj/landmark/corpse/miner, /obj/landmark/corpse/miner/rig, /obj/landmark/corpse/bridgeofficer, /obj/landmark/corpse/commander, \
 	/obj/landmark/corpse/russian)
-	new A
\ No newline at end of file
+	new A
+
+/obj/structure/closet/coffin/attack_hand(mob/user)
+	add_fingerprint(user)
+	if(opened)
+		to_chat(user, SPAN_NOTICE("You can't fit the cover back on without hammering it into place!"))
+	else
+		to_chat(user, SPAN_NOTICE("The cover is too heavy to lift without a prying tool!"))
+
+/obj/structure/closet/coffin/proc/burn()
+	add_overlay("coffin_pyre")
+	on_fire = TRUE
+	burning = addtimer(CALLBACK(src, PROC_REF(pyre), src), 120 SECONDS, TIMER_STOPPABLE) //TODO: Add TIMER_STOPPABLE to being extinguished, which involves giving reagents touch effects to structures, which they currently don't affect
+
+/obj/structure/closet/coffin/attackby(obj/item/I, mob/user)
+	if(!on_fire && isflamesource(I))
+		user.visible_message(SPAN_WARNING("[user] has lit the [src] on fire! In a couple minutes, it and its occupant will be ash!"))
+		burn()
+
+	var/list/usable_qualities = list()
+	if(opened)
+		usable_qualities += QUALITY_SAWING
+		usable_qualities += QUALITY_PRYING
+		usable_qualities += QUALITY_HAMMERING
+	if(!opened)
+		usable_qualities += QUALITY_PRYING
+
+	var/tool_type = I.get_tool_type(user, usable_qualities, src)
+	switch(tool_type)
+		if(QUALITY_PRYING)
+			if(!opened)
+				if(I.use_tool(user, src, WORKTIME_FAST, tool_type, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					visible_message(
+							SPAN_NOTICE("\The [src] has been pried open by [user] with \the [I]."),
+							"You hear [tool_type]."
+					)
+					open()
+			else
+				if(I.use_tool(user, src, WORKTIME_FAST, tool_type, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					visible_message(
+							SPAN_NOTICE("\The [src] has been pried apart by [user] with \the [I]."),
+							"You hear [tool_type]."
+					)
+					drop_materials(drop_location())
+					qdel(src)
+			return
+
+		if(QUALITY_SAWING)
+			if(opened)
+				if(I.use_tool(user, src, WORKTIME_FAST, tool_type, FAILCHANCE_EASY, required_stat = STAT_MEC))
+					visible_message(
+							SPAN_NOTICE("\The [src] has been cut apart by [user] with \the [I]."),
+							"You hear [tool_type]."
```

Comment:
"You hear sawing."
"You hear hammering."
"You hear prying."
Sound messages derived from tool qualities sound odd. How about:
"You hear wood sawing."
"You hear nails being pried out."
"You hear nails being hammered in."
Something like that.

Reply:
I *KNOW* it's already like that in some messages. Be better, put some effort.